### PR TITLE
Timeout tests after 90s (default)

### PR DESCRIPTION
**Pull request by marcoderama**
Added a `timeout` function for running tests.  Default is 90s.  If nothing times out, then behavior is unchanged.  If there's a timeout, then the FAILURES are incremented, and also tracked as a timeout failure.

For example, I hacked the following change to `smokecommon`:
```
@@ -152,10 +152,11 @@ test_es_console_responding() {
 # Grafana access test
 test_grafana_responding() {
   grafana_responding() {
+      sleep 4
       results=$(curl_query $1 "api/org" | jq '.id')
       ! [[ $results -eq 0 ]]
   }
-  timeout grafana_responding "$@"
+  timeout 1 grafana_responding "$@"
   T $? grafana service accessible via $1
 }
```

That forced the `test_grafana_responding` test to take at least 4 seconds.  Note also that I specified a non-default timeout of 1 second for this test.  The output from running `smoke_services` looked like:

```
PASS: Lightbend Console accessible via http://192.168.99.100:30080
*** TIMEOUT: grafana service accessible via http://192.168.99.100:30030
*** FAIL: grafana service accessible via http://192.168.99.100:30030
./smoke_services: line 23: 84374 Terminated: 15          "$@"
PASS: prometheus service accessible via http://192.168.99.100:30090
PASS: alertmanager accessible via http://192.168.99.100:30093
*** TIMEOUT: grafana service accessible via http://192.168.99.100:30080/service/grafana
*** FAIL: grafana service accessible via http://192.168.99.100:30080/service/grafana
./smoke_services: line 27: 84392 Terminated: 15          "$@"
PASS: prometheus service accessible via http://192.168.99.100:30080/service/prometheus
PASS: es-monitor-api service accessible via http://192.168.99.100:30080/service/es-monitor-api
Some smoke_services tests failed!  PASS: 5  FAIL: 2  (TIMEOUTS: 2)
```

### CODE REVIEW

File: enterprise-suite/tests/timeout.fn [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# Execute a command with a timeout
+
+# This has been modified so that it's a function instead of a command.
+# Also return 142 on timeout to match "kill SIGALARM" return value.
+
+# License: LGPLv2
+# Author:
+#    http://www.pixelbeat.org/
+# Notes:
+#    Note there is a timeout command packaged with coreutils since v7.0
```
Comment (jsravn) [inline enterprise-suite/tests/timeout.fn]: @marcoderama is there a reason not to use the coreutils timeout command?

File: enterprise-suite/tests/timeout.fn [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+a=
+
+timeout_cleanup()
+{
+	trap - ALRM                #reset handler to default
+    kill -ALRM $a 2>/dev/null  #stop timer subshell if running
+    last=$!
+	kill $last 2>/dev/null &&  #kill last job
+      return 142               #exit with 142 if it was running
+}
```
Comment (jsravn) [inline enterprise-suite/tests/timeout.fn]: Formatting seems messed up here.

File: enterprise-suite/tests/timeout.fn [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+timeout_watchit()
+{
+    trap "timeout_cleanup" ALRM
+	sleep $1 & wait
+	kill -ALRM $$
+}
```
Comment (jsravn) [inline enterprise-suite/tests/timeout.fn]: Same - `trap` line is not indented the same.

File: enterprise-suite/tests/timeout.fn (hunk 1 above)
Comment (mitkus) [reply enterprise-suite/tests/timeout.fn]: Can only depend on coreutils on linux. Mac OS, for example, doesn't have `timeout`.

File: enterprise-suite/tests/smokecommon [hunk 4]
```diff
@@ -32,8 +32,37 @@ busy_wait() {
...
+exec 7<$FAILURES
+# unlink so we don't have to clean up later
+rm $FAILURES
+
+TIMEOUTS=$(mktemp -t pipe.XXXXX)
+exec 8>$TIMEOUTS
+exec 9<$TIMEOUTS
+# unlink so we don't have to clean up later
+rm $TIMEOUTS
+
```
Comment (jsravn) [inline enterprise-suite/tests/smokecommon]: Would this be simpler if we used `mkfifo` instead? Then we can write/read from the named pipes rather than anonymous file descriptors, which I find pretty hard to follow.

Comment (marcoderama) [reply enterprise-suite/tests/smokecommon]: I originally tried with mkfifo but it just wouldn't work.  I would close it for writing but that would close it for reading at the same time and it would disappear.  I tried various stunts but all to no avail.  It just worked as soon as I moved to an actual file.

I also tried to use variables for the file descriptors but that got even uglier fast.  (cf. https://stackoverflow.com/questions/8295908/how-to-use-a-variable-to-indicate-a-file-descriptor-in-bash)  And I skipped 5 cuz some page suggested that was a good idea...

So it is what it is.

**Comment by marcoderama** [comment]
And yeah @jsravn, I think we've pushed the shell scripting as far as we want to here.
